mpi_requirements.txt

```diff
@@ -0,0 +1 @@
+mpi4py
```

Comment:
I really think we should generalise this and rather call it ``requirements-optional.txt`` or something along these lines. This is because we have several non-compulsory packages that we may want to mark as optional

Reply:
Optional requirements should be split into individual purpose in my opninion, installing all optional if you only want mpi for example is overkill. Moving everything to an ptional file would result in the same thing we have currently

Reply:
in a big project you can have 20 optional packages, would you then keep ~20 different files, one for each of them? 

> installing all optional if you only want mpi for example is overkill.
splitting into compulsory+optional is a very good thing already; we cannot clearly support all possible installation combinations. 
* On CI, it would make sense to either install all optionals or none;
* If users don't want a specific subset of packages, then it's up to them. For example, instead of using conda, they could use the setup.py based installation and only install a specific subset of the desired features (see below for my other comment about setup.py)

https://stackoverflow.com/questions/3664478/optional-dependencies-in-a-pip-requirements-file

Reply:
I mean sure, but this PR is not about this, this is only about making sure we don't break everyone's code forcing MPI.

Reply:
doesn't matter, we should do it properly